resources/list.py

```diff
@@ -1,34 +1,224 @@
-
-    # Get ListItems (GET api/users/<user_id>/lists/<list_id>/items)
-        # Get all items & details for user & list
-    # Get ListItem by item_Id and list_id (GET api/users/<user_id>/lists/<list_id>/items?=item_id=)
-    @staticmethod
-    def get():
-        pass
-
-    # Add Item to List
-        # Create a ListItem
-    @staticmethod
-    def post():
-        pass
-
-    # remove item from list
-
-# class ListItemResource(Resource):
-# /api/lists/<list_id:int>/items
-# Delete
-# Post
\ No newline at end of file
+    # Create a List - linked to user who owns it and a group they're in
+    @jwt_required()
+    def post(self):
+        """
+        Create a new List associated to user who owns it & a group
+        params:
+        - owned_by_user: int id of user - required
+        - group_id: int id of a group
+        - name:
+        returns:
+        - user does not exist: 400
+        - user not a member of the group: 400
+        - success: 200 and serialized List
+        """
+        try:
+            post_args = reqparse.RequestParser()
+            post_args.add_argument('owned_by_user', type=int, required=True)
+            post_args.add_argument('group_id', type=int)
+            post_args.add_argument('name')
+            args = post_args.parse_args()
+
+            if not User.exists(args.owned_by_user):
+                return {'message': f'User with id {args.owned_by_user} does not exist'}, 400
+
+            if not Group.exists(args.group_id):
+                return {'message': f'Group with id {args.group_id} does not exist'}, 400
+
+            user = User.get(id=args.owned_by_user)
+            group = Group.get(id=args.group_id)
+
+            if user not in group.users:
+                return {'message': f'User with id {args.user_id} is not in group'}, 400
+
+            list_id = List.create(owned_by_user=args.owned_by_user,
+                                  group_id=args.group_id,
+                                  name=args.name)
+
+            list = List.get(id=list_id)
+
+            return serialize(list), 200
+
+        except Exception as ex:
+            return {'message': 'An internal service error occurred', 'error': str(ex)}
+
+    @jwt_required()
+    def put(self):
+        """
+        Update List information
+        returns:
+        - list does not exist: 400
+        - success: 200 and serialized List
+        """
+        try:
+            put_args = reqparse.RequestParser()
+            put_args.add_argument('id', type=int, required=True)
+            put_args.add_argument('owned_by_user', type=int)
+            put_args.add_argument('name')
+            args = put_args.parse_args()
+
+            if not List.exists(args.id):
+                return {'message': f'List with id {args.id} does not exist'}, 400
+
+            if args.owned_by_user is not None and not User.exists(args.owned_by_user):
+                return {'message': f'User with id {args.owned_by_user} does not exist'}, 400
```

Comment:
I think we need that check because 'owned_by_user' may not necessarily be passed in on an update request, but the .add_argument will put it as None, and then you'll be checking for User.exists(None) which will make it mad.